authentication/mtls.go

```diff
@@ -143,6 +147,12 @@ func (a MTLSAuthenticator) Middleware() Middleware {
 	}
 }
 
+func (a MTLSAuthenticator) GRPCMiddleware() grpc.StreamServerInterceptor {
+	return grpc_middleware_auth.StreamServerInterceptor(func(ctx context.Context) (context.Context, error) {
```

Comment:
Should we not support mTLS authentication as well? Or you'd like to this in a separate PR? Maybe we could at least at a to-do.